src/Socket/Socket.cpp

```diff
@@ -68,3 +66,168 @@ Socket::set_socket_non_blocking()
+	{
+		delete_handler();
+	}
+	_response.load_http_request(_request);
+	clean_request();
+	send_response();
+	return 0;
+}
+
+void
+Socket::multipart_handler()
+{
+	ssize_t		  byte_read;
+	char		  buffer[MAXLINE] = { 0 };
+	unsigned long content_length = std::stoul(_request._request_map["Content-Length"]);
+
+	while (_body_str.size() < content_length)
+	{
+		std::memset(buffer, 0, MAXLINE);
+		byte_read = recv(_connection_fd, buffer, MAXLINE - 1, 0);
+		_body_str.append(buffer, byte_read);
+	}
+	if (LOG_SOCKET)
+		std::cout << "_body_str.size(): " << _body_str.size() << std::endl;
+	create_new_file();
+}
+
+void
+Socket::delete_handler()
+{
+	std::string file_name = _request.get_path();
+
+	std::string fullpath = _dir_path + "/uploads" + file_name;
+	if (access(fullpath.c_str(), F_OK) != -1)
+	{
+		_request._request_map["FileName"] = "exist";
+		int ret = remove(fullpath.c_str());
+		if (ret != 0)
+			_request._request_map["FileName"] = "r_fail";
+	}
+}
+
+void
+Socket::socket_accept()
+{
+	_connection_fd = accept(get_sock_id(), NULL, NULL);
+	if (_connection_fd < 0)
+	{
+		std::cout << "connection_fd: " << _connection_fd << " Failed!" << std::endl;
+	}
+}
+
+std::string
+Socket::get_file_full_name()
+{
+	std::string start_looking = _body_str.substr(_body_str.find("filename="));
+	start_looking = start_looking.substr(start_looking.find_first_of('"'), start_looking.find("\n"));
+	size_t		position_quote_start(start_looking.find_first_of('"') + 1);
+	size_t		length(start_looking.find_first_of('"', +1) - position_quote_start);
+	std::string file_name = start_looking.substr(position_quote_start, length);
+	std::string fullpath = "test/website/uploads/" + file_name;
+	return fullpath;
+}
+
+void
+Socket::create_new_file()
+{
+	std::string fullpath = get_file_full_name();
+	if (LOG_SOCKET)
+		std::cout << "new file: \"" << fullpath << "\"" << std::endl;
+	if (access(fullpath.c_str(), F_OK))
+	{
+		size_t		delimiter = _body_str.find("Content-Type");
+		std::string file_part
+			= _request.trim(_body_str.substr(_body_str.find_first_of("\r\n\r\n", delimiter)));
+		if (LOG_SOCKET)
+			std::cout << "file_part: " << file_part << std::endl;
+		size_t		end = file_part.find(_request._request_map["boundary"]);
+		std::string half_clean_file = file_part.substr(0, end);
+		std::string clean_file = clean_end_of_file(half_clean_file);
+		if (LOG_SOCKET)
+			std::cout << "### clean file ###\n" << clean_file << "\n###/clean file ###\n" << std::endl;
+		std::ofstream ofs(fullpath.c_str(), std::ios_base::out | std::ios_base::binary);
+		ofs.write(clean_file.c_str(), clean_file.size() - 1);
+		ofs.close();
+	}
+	else
+	{
+		_request._request_map["FileName"] = "exist";
+	}
+}
+
+std::string
+Socket::clean_end_of_file(std::string const &str_to_clean)
+{
+	int index = str_to_clean.size() - 1;
+	while (str_to_clean[index] == '-' || str_to_clean[index] == '\n')
+		--index;
+	return str_to_clean.substr(0, index);
+}
+
+std::string
+Socket::get_dir_path() const
```

Comment:
Is it possible to limit as much as possible ?
```suggestion
const std::string &
Socket::get_dir_path() const
```

Reply:
done 29d4a1f12eb125e5c681dd4d345ac48b5d26bb7c

Reply:
and the declaration:
cf1d943d6bcc2168cb8d431d4d2c2d92ad56b661